### ISSUE TITLE
#2205 breaks make html for definition list with classifiers that contains regular-expression like string

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#2247]
It caused by #2205.

The issue is reported on https://groups.google.com/d/msg/sphinx-users/nH3XTex6cY4/AtEdUQtKDwAJ

```
Hello everyone,

This is an odd one.  I/we've been using Sphinx for our docs for a longtime.  I just tried to upgrade to 1.3.4 release, and I got an error from a restructured text "definition list" with a square bracket (this definition caused no problems with Sphinx 1.3.3 and earlier).  Here is more details:

Text that causes error
================

outputformatlist : outputFormatObj[]
    Array of the available output formats.

Error during "make html"
==================

 reading sources... [ 73%] mapscript/mapscript
Exception occurred:
  File "C:\Python27\lib\re.py", line 251, in _compile
    raise error, v # invalid expression
error: unexpected end of regular expression
The full traceback has been saved in d:\temp\sphinx-err-cyh44l.log, if you want to report
the issue to the developers.
Please also report this if it was a user error, so that a better error message can be prov
ided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>.
 Thanks!

Traceback
========

# Sphinx version: 1.3.4
# Python version: 2.7.11 (CPython)
# Docutils version: 0.12 release
# Jinja2 version: 2.8
# Last messages:
#   reading sources... [ 71%] mapfile/template
#   reading sources... [ 71%] mapfile/union
#   reading sources... [ 71%] mapfile/web
#   reading sources... [ 71%] mapfile/xml_mapfile
#   reading sources... [ 72%] mapfile/xmp_metadata
#   reading sources... [ 72%] mapscript/imagery
#   reading sources... [ 72%] mapscript/index
#   reading sources... [ 73%] mapscript/introduction
#   reading sources... [ 73%] mapscript/mapfile
#   reading sources... [ 73%] mapscript/mapscript
# Loaded extensions:
#   alabaster (0.7.6) from C:\Python27\lib\site-packages\alabaster-0.7.6-py2.7.egg\alabaster\__init__.pyc
#   labels (unknown version) from D:\build\mapserver-docs-git-branch-7-0\labels.pyc
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\sphinx-1.3.4-py2.7.egg\sphinx\cmdline.py", line 244, in main
    app.build(opts.force_all, filenames)
  File "C:\Python27\lib\site-packages\sphinx-1.3.4-py2.7.egg\sphinx\application.py", line 266, in build
    self.builder.build_update()
  File "C:\Python27\lib\site-packages\sphinx-1.3.4-py2.7.egg\sphinx\builders\__init__.py", line 251, in build_update
    'out of date' % len(to_build))
  File "C:\Python27\lib\site-packages\sphinx-1.3.4-py2.7.egg\sphinx\builders\__init__.py", line 265, in build
    self.doctreedir, self.app))
  File "C:\Python27\lib\site-packages\sphinx-1.3.4-py2.7.egg\sphinx\environment.py", line 618, in update
    self._read_serial(docnames, app)
  File "C:\Python27\lib\site-packages\sphinx-1.3.4-py2.7.egg\sphinx\environment.py", line 638, in _read_serial
    self.read_doc(docname, app)
  File "C:\Python27\lib\site-packages\sphinx-1.3.4-py2.7.egg\sphinx\environment.py", line 791, in read_doc
    pub.publish()
  File "C:\Python27\lib\site-packages\docutils-0.12-py2.7.egg\docutils\core.py", line 218, in publish
    self.apply_transforms()
  File "C:\Python27\lib\site-packages\docutils-0.12-py2.7.egg\docutils\core.py", line 199, in apply_transforms
    self.document.transformer.apply_transforms()
  File "C:\Python27\lib\site-packages\docutils-0.12-py2.7.egg\docutils\transforms\__init__.py", line 171, in apply_transforms
    transform.apply(**kwargs)
  File "C:\Python27\lib\site-packages\sphinx-1.3.4-py2.7.egg\sphinx\transforms.py", line 182, in apply
    apply_source_workaround(n)
  File "C:\Python27\lib\site-packages\sphinx-1.3.4-py2.7.egg\sphinx\util\nodes.py", line 53, in apply_source_workaround
    '\s*:\s*%s' % classifier.astext(), '', node.rawsource)
  File "C:\Python27\lib\re.py", line 155, in sub
    return _compile(pattern, flags).sub(repl, string, count)
  File "C:\Python27\lib\re.py", line 251, in _compile
    raise error, v # invalid expression
error: unexpected end of regular expression


****

Anyone have any ideas why this "[]" causes this problem, and only with the 1.3.4 release?

Thanks,

-jeff
www.mapserver.org
```

**Comment by tk0miya** [comment]
I fixed at 34cbf3b.
@shimizukawa Could you review it to make sure?

**Comment by shimizukawa** [comment]
LGTM. Thanks!
